aicsimageio/readers/mrc_reader.py

```diff
@@ -0,0 +1,114 @@
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with mrcfile.open(path, permissive=True, header_only=True):
+                return True
+        except (ValueError, TypeError):
+            return False
+
+    def __init__(self, image: types.PathLike):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        # Catch non-local file system
+        if not isinstance(self._fs, LocalFileSystem):
+            raise NotImplementedError(
+                f"mrc reader not yet implemented for non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            # need full mmap open here because `is_image_stack` needs access to data shape
+            with mrcfile.mmap(self._path, permissive=True) as mrc:
+                if mrc.is_image_stack():
+                    n_scenes = mrc.header.nz
+                else:
+                    n_scenes = mrc.header.nz / mrc.header.mz
+                self._scenes = [f'Image:{i}' for i in range(int(n_scenes))]
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+        return self._make_xarray(delayed=True)
+
+    def _read_immediate(self) -> xr.DataArray:
+        return self._make_xarray(delayed=False)
+
+    def _make_xarray(self, delayed: bool) -> xr.DataArray:
+        with mrcfile.mmap(self._path, permissive=True) as mrc:
+            if mrc.is_single_image():
+                scene = mrc.data
+                dims = [DimensionNames.SpatialY, DimensionNames.SpatialX]
+            elif mrc.is_image_stack():
+                scene = mrc.data[self._current_scene_index]
+                dims = [DimensionNames.SpatialY, DimensionNames.SpatialX]
+            elif mrc.is_volume():
+                scene = mrc.data
+                dims = [DimensionNames.SpatialZ,
+                        DimensionNames.SpatialY,
+                        DimensionNames.SpatialX]
+            else:
+                scene = mrc.data[self._current_scene_index]
+                dims = [DimensionNames.SpatialZ,
+                        DimensionNames.SpatialY,
+                        DimensionNames.SpatialX]
+
+            # convert before exiting the context manager or scene will be None
+            if delayed:
+                data = da.from_array(scene)
+            else:
+                data = np.asarray(scene)
+
+        return xr.DataArray(data, dims=dims)
```

Comment:
coords and metadata?

coords are the coordinate vectors for each dimension. see this function for an example: https://github.com/AllenCellModeling/aicsimageio/blob/f3d8f9d474e791d8e1544addbc88d0802f19aae2/aicsimageio/metadata/utils.py#L601

they allow for additional data selection: https://github.com/AllenCellModeling/aicsimageio#xarray-coordinate-plane-attachment

take your physical pixel sizes and use this function: https://github.com/AllenCellModeling/aicsimageio/blob/f3d8f9d474e791d8e1544addbc88d0802f19aae2/aicsimageio/readers/reader.py#L729

Should be good to go.

Metadata on the other hand... Is there no metadata at all?

Reply:
Added coordinates! There is some metadata; I was just not sure which and how to add. For now, I followed the pattern of other readers and simply passed the parsed `header` of the file as `constants.METADATA_UNPROCESSED`.